.github/workflows/snyk-security.yaml

```diff
@@ -0,0 +1,41 @@
+name: Snyk Security
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  snyk-scan:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+            submodules: recursive
+
+      - name: Install Snyk CLI
+        run: npm install -g snyk
+
+      - name: Snyk Monitor and Test multiple projects
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+          SNYK_ORG: ${{ secrets.SNYK_ORG }}
+        run: |
+          echo "Fetching tags"
+          git fetch origin 'refs/tags/*:refs/tags/*'
+
+          echo "Authenticating with Snyk"
+          snyk auth ${SNYK_TOKEN}
+
+          echo "Scanning project: codeflare-sdk/main"
+          snyk monitor --all-projects --exclude=requirements.txt --org=${SNYK_ORG} --target-reference="main"
+
+          # This list is based off RHOAI Supported versions: https://access.redhat.com/support/policy/updates/rhoai-sm/lifecycle
+          # Compared to the tags in the ImageStream annotations: https://github.com/red-hat-data-services/notebooks/blob/rhoai-2.8/manifests/base/jupyter-datascience-notebook-imagestream.yaml
+          # Loop through the list of released tags and scan each project
+          list_of_released_tags=("v0.22.0" "v0.21.1" "v0.19.1", "v0.16.4", "vv0.14.1")
```

Comment:
Is the `vv` here on purpose?

Reply:
Yes, that's not a typo. It's the official release tag

Reply:
how will we maintain the list up to date?

Reply:
Good question, in terms of _new release tags_ the release workflow will add all subsequent new release tags to the list, this way monitoring all newest tags.

In terms of older tags, RHOAI releases and their support period are listed [[here]](https://access.redhat.com/support/policy/updates/rhoai-sm/lifecycle). Sometimes, the support can be extended for some of the versions. It's best to manually remove unused tags from time to time. There's no harm in monitoring multiple tags.